### PR TITLE
[WIP] change default query end time to math.MaxInt

### LINKED ISSUE #5047: Default query end time should be math.MaxInt64
Too many people have reported bugs where they can't see their data because they wrote a data point with a future timestamp. This is because by default we set the query end time to `time.Now()` if they didn't specify an end time.

We should have the end time default to `time.Unix(0, math.MaxInt64)`

### PR DESCRIPTION
**Pull request by Wind0r**
This should fix #5047 but tests still fail.
I dont know why.

### CODE REVIEW

**Comment by otoolep** [comment]
Why do they fail? What is the test output?

**Comment by otoolep** [comment]
Have you signed the CLA @Wind0r ?

File: influxql/ast.go [hunk 1]
```diff
@@ -2954,7 +2955,7 @@ func TimeRangeAsEpochNano(expr Expr) (min, max int64) {
 		min = tmin.UnixNano()
 	}
 	if tmax.IsZero() {
-		max = time.Now().UnixNano()
+		max = time.Unix(0, math.MaxInt64).UnixNano()
```
Comment (otoolep) [inline influxql/ast.go]: The function's comment also needs to be updated.

**Comment by Wind0r** [comment]
Yes i signed CLA.

Testoutput:
 ```
--- FAIL: TestServer_Query_NoTimeBounds (0.24s)
	server_test.go:1006: select without time bound should return all data: unexpected results
		query:  SELECT * FROM db0.rp0.cpu
		exp:    {"results":[{"series":[{"name":"cpu","columns":["time","host","value"],"values":[["1970-01-01T00:00:00.000000001Z","server01",1],["2262-04-11T23:47:16.854775806Z","server01",2]]}]}]}
		actual: {"results":[{"series":[{"name":"cpu","columns":["time","host","value"],"values":[["1970-01-01T00:00:00.000000001Z","server01",1]]}]}]}

FAIL
```

The other value is still missing :(

**Comment by otoolep** [comment]
Hmmm -- I see that @Wind0r -- looks like this may not be as simple as I thought! Do you get the right data out if you:

a) pass in a the maximum value -1
b) pass in an upper bound to the SELECT that should include the data?

**Comment by Wind0r** [comment]
a) changing it to max-1 or max-X doesn't help (both ast and test)
b) mhh.  also do not help but idk if i am doing it correct.

using time.Now() as test input works well.. but even adding 1 Hour to it let it fail.

Maybe we got problems with the max range?
https://influxdb.com/docs/v0.9/troubleshooting/frequently_encountered_issues.html#querying-outside-the-min-max-time-range

I do not have any time left. i will watch over it tomorrow again.

**Comment by Wind0r** [comment]
Sorry for the late response.

Executing the Query with 'where time <= ''MaxInt64'' works well.
I setup a test with and one without the where clause and watched the ast.go:TimeRangeAsEpochNano return values.
In both setups the function returns the same values, but only the query with where returns all values and passes the test.

Any hint where also need to change the code?

**Comment by otoolep** [comment]
@Wind0r -- I don't know of anywhere else in the code that could have an impact on this.

**Comment by jsternberg** [comment]
I think we do want a change like this at some point, but this change is too old to merge and the tests are failing.

Thank you for your contribution. If you rebase and get this working, feel free to reopen the PR and someone will review it.

**Pull request closed without merging**